RecoEgamma/PhotonIdentification/plugins/PhotonIDValueMapProducer.cc

```diff
@@ -212,9 +212,8 @@ void PhotonIDValueMapProducer::produce(edm::StreamID, edm::Event& iEvent, const
     std::vector<float> vars[nVars_];
 
     // reco::Photon::superCluster() is virtual so we can exploit polymorphism
-    for (unsigned i = 0; i < src->size(); ++i) {
-        const auto& iPho = src->ptrAt(i);
-
+    for(auto const& iPho : src->ptrs())
```

Comment:
By quick look this loop could also avoid `edm::Ptr`
```suggestion
    for(auto const& pho : *src)
```